libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ProbabilityDistributionBoxChart.tsx

```diff
@@ -30,39 +30,21 @@ export class ProbabilityDistributionBoxChart extends React.Component<IProbabilit
   public context: React.ContextType<typeof ModelAssessmentContext> =
     defaultModelAssessmentContext;
 
+  public componentDidMount(): void {
+    this.calculateBoxDataIfNeeded();
+  }
+
   public componentDidUpdate(
     prevProps: IProbabilityDistributionBoxChartProps
   ): void {
-    if (
-      this.props.boxPlotState.boxPlotData.length === 0 ||
-      !_.isEqual(prevProps.selectedCohorts, this.props.selectedCohorts) ||
-      !_.isEqual(
-        prevProps.probabilityOption?.id,
-        this.props.probabilityOption?.id
-      )
-    ) {
-      const boxPlotData = this.props.selectedCohorts.map(
-        (cohort: ErrorCohort, index: number) => {
-          return calculateBoxPlotDataFromErrorCohort(
-            cohort,
-            index,
-            this.props.probabilityOption?.key || "",
-            this.props.probabilityOption?.text,
-            this.context.requestBoxPlotDistribution,
-            ifEnableLargeData(this.context.dataset)
-          );
-        }
-      );
-      setOutlierDataIfChanged(
-        boxPlotData,
-        this.props.boxPlotState,
-        this.props.onBoxPlotStateUpdate
-      );
-    }
+    this.calculateBoxDataIfNeeded(prevProps);
   }
 
   public render(): React.ReactNode {
     const theme = getTheme();
+    if (this.props.boxPlotState.boxPlotData.length === 0) {
```

Comment:
nit: if(!this.props.boxPlotState.boxPlotData.length) {...}, same for the if statement below

Reply:
Indeed. As long as we don't have the box plot data we can't display anything. The other if is in a separate method, though. The box plot data calculation may end up getting rather expensive with larger datasets, which is why I want to be careful when calling it and avoid unnecessarily calculating it.

Reply:
Ah you meant the === 0 part! Done!